### DISCUSSION TITLE
Build a User Management App with SvelteKit help

### DISCUSSION THREAD

**Discussion by berpel** [supabase/supabase#26272] [Questions] [answered]
I am following the guide for [Build a User Management App with SvelteKit help](https://supabase.com/docs/guides/getting-started/tutorials/with-sveltekit?queryGroups=language&language=ts). Everything worked great and was able to get the magic link delivered and the callback working and session being created, but on the account page I am getting the following error:

```
Internal server error: /Users/brett/code/sessions/src/routes/account/+page.svelte:4:13 Unexpected token
  Plugin: vite-plugin-svelte
  File: /Users/brett/code/sessions/src/routes/account/+page.svelte:4:13
   2 |  <script lang="ts">
   3 |    import { enhance } from '$app/forms';
   4 |    import type { SubmitFunction } from '@sveltejs/kit';
                      ^
   5 |    //import Avatar from './Avatar.svelte'
   6 |   (x2)
{
  name: 'ParseError',
  id: '/Users/brett/code/sessions/src/routes/account/+page.svelte',
  message: '/Users/brett/code/sessions/src/routes/account/+page.svelte:4:13 Unexpected token',
  frame: ' 2 |  <script lang="ts">\n' +
    " 3 |    import { enhance } from '$app/forms';\n" +
    " 4 |    import type { SubmitFunction } from '@sveltejs/kit';\n" +
    '                    ^\n' +
    " 5 |    //import Avatar from './Avatar.svelte'\n" +
    ' 6 |  ',
  code: 'parse-error',
  stack: '',
  loc: {
    line: 4,
    column: 13,
    file: '/Users/brett/code/sessions/src/routes/account/+page.svelte'
  },
  plugin: 'vite-plugin-svelte',
  pluginCode: '<!-- src/routes/account/+page.svelte -->\n' +
```

Has anyone else come across this and know of a solution?

Thanks.

**Comment by silentworks** [comment]
No I have never seen this issue before. Which version of SvelteKit are you using and also which version of the `vite-plugin-svelte` are you using too? From the error you pasted it looks like there might be an invisible new line in the code, did you copy and paste it from the tutorial? if yes then try and type line 4 out manually and see if you get the same error.

**Reply by berpel** [reply]
Hi, thanks for the reply. I thought there might be a weird character also so I had already tried to manually type it out and still got the same thing.

Here are my `package.json` dependencies:

```	"devDependencies": {
		"@sveltejs/adapter-auto": "^3.0.0",
		"@sveltejs/kit": "^2.0.0",
		"@sveltejs/vite-plugin-svelte": "^3.0.0",
		"@types/eslint": "^8.56.0",
		"eslint": "^8.56.0",
		"eslint-config-prettier": "^9.1.0",
		"eslint-plugin-svelte": "^2.35.1",
		"prettier": "^3.1.1",
		"prettier-plugin-svelte": "^3.1.2",
		"svelte": "^4.2.7",
		"svelte-check": "^3.6.0",
		"typescript": "^5.0.0",
		"vite": "^5.0.3"
	},
	"type": "module",
	"dependencies": {
		"@supabase/auth-ui-shared": "^0.1.8",
		"@supabase/auth-ui-svelte": "^0.2.9",
		"@supabase/ssr": "^0.3.0",
		"@supabase/supabase-js": "^2.43.1"
	}
```

Also, if i comment out that line it then throws an error here:
```
  Internal server error: /Users/brett/code/sessions/src/routes/account/+page.svelte:13:16 Unexpected token
  Plugin: vite-plugin-svelte
  File: /Users/brett/code/sessions/src/routes/account/+page.svelte:13:16
   11 |    $: ({ session, supabase, profile } = data)
   12 |  
   13 |    let profileForm: HTMLFormElement
                          ^
   14 |    let loading = false
   15 |    let fullName: string = profile?.full_name ?? ''
```

I am using node v20.6

**Reply by silentworks** [reply]
This seems to be a typescript issue, so I would assume the `@svelte/vite-plugin-svelte` isn't working correctly. I'd say delete your `node_modules` and `.svelte-kit` directories and then try installing the node modules again.

I'm on node v20.11.0 and don't have this issue.

**Reply by berpel** [reply]
I appreciate all your help. I've tried your suggestion and I am still getting the same error.

I even went as far as creating a new project and following the guide again and I am having the same issue. I've also upgraded to the latest version of Node v20.

**Reply by berpel** [reply]
I've created a repo of my code if you could pull it and see if it works for you that would be amazing.

https://github.com/berpel/sveltekit-supabase

**Reply by silentworks** [reply, marked as answer]
You are missing the preprocessor for sveltekit that is necessary to compile the Typescript code. Your `svelte.config.js` file should look like:

```js
import adapter from '@sveltejs/adapter-auto';
import { vitePreprocess } from '@sveltejs/vite-plugin-svelte';

/** @type {import('@sveltejs/kit').Config} */
const config = {
	// Consult https://kit.svelte.dev/docs/integrations#preprocessors
	// for more information about preprocessors
	preprocess: vitePreprocess(),

	kit: {
		// adapter-auto only supports some environments, see https://kit.svelte.dev/docs/adapter-auto for a list.
		// If your environment is not supported or you settled on a specific environment, switch out the adapter.
		// See https://kit.svelte.dev/docs/adapters for more information about adapters.
		adapter: adapter()
	}
};

export default config;
```

**Reply by berpel** [reply]
Thank you so much for all your help!
